interfaces/acados_template/acados_template/utils.py

```diff
@@ -232,6 +231,24 @@ def get_shared_lib_prefix():
     else:
         return 'lib'
 
+def get_binary_ext():
+    if os.name == 'nt':
+        return '.exe'
+    else:
+        return ''
+
+def get_architecture_amd64_arm64():
+    # common uname -m results
+    # https://en.wikipedia.org/wiki/Uname
+    current_arch = platform.machine()
+    amd64_compatible = ["i3", "i6", "amd", "x86"]
+    arm64_compatible = ["arm", "aarch"]
+    if any([current_arch.lower().startswith(arch) for arch in amd64_compatible]):
+        return "amd64"
+    elif any([current_arch.lower().startswith(arch) for arch in arm64_compatible]):
+        return "arm64"
+    return None
```

Comment:
I could `raise Exception` here if that's more in-line with the codebase or this helper could be use elsewhere. If it's just here then I think throwing and immediately catching and `sys.exit()` is a little bit silly.